api/client/swarm/opts.go

```diff
@@ -99,27 +101,31 @@ func (o *AutoAcceptOption) String() string {
 }
 
 // Set sets a new value on this option
-func (o *AutoAcceptOption) Set(value string) error {
-	value = strings.ToUpper(value)
-	switch value {
-	case "", "NONE":
-		if accept, ok := o.values[WORKER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", WORKER)
-		}
-		if accept, ok := o.values[MANAGER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", MANAGER)
-		}
-		o.values[WORKER] = false
-		o.values[MANAGER] = false
-	case WORKER, MANAGER:
-		if accept, ok := o.values[value]; ok && !accept {
-			return fmt.Errorf("value NONE is incompatible with %s", value)
+func (o *AutoAcceptOption) Set(acceptValues string) error {
+	for _, value := range strings.Split(acceptValues, ",") {
+		value = strings.ToUpper(value)
+		switch value {
+		case "", NONE:
+			// NONE must stand alone, so if any non-NONE setting exists, conflict
+			if !o.values[NONE] && len(o.values) > 0 {
+				var s []string
+				for r := range o.values {
+					s = append(s, r)
+				}
+				return fmt.Errorf("value NONE is incompatible with %s", strings.Join(s, ", "))
+			}
+			o.values[WORKER] = false
+			o.values[MANAGER] = false
```

Comment:
Thanks for the feedback Aaron. Can you elaborate a little more on why this change would be more beneficial? (it was `map[string]bool` before I got to it, so just want to understand the benefits). Thanks!